### ISSUE TITLE
[distributed] Race detected in TestListenAndServePlain 

### ISSUE THREAD

**Issue by vadmeste** [minio/minio#3175]
Running go test with count=100 detect more possible races:
`go test -run=TestListenAndServePlain -race -count=100`

```
ERRO[0000] Unable to serve incoming requests.            cause=accept tcp 127.0.0.1:42456: use of closed network connection location=[server-mux.go:324:(*ServerMux).List
enAndServe.func1()]
==================
WARNING: DATA RACE
Write at 0x00000176cdc8 by goroutine 25:
  github.com/minio/minio/cmd.TestListenAndServePlain()
      /home/vadmeste/work/gospace/src/github.com/minio/minio/cmd/server-mux_test.go:225 +0x1ac
  testing.tRunner()
      /home/vadmeste/work/go/src/testing/testing.go:610 +0xc9

Previous read at 0x00000176cdc8 by goroutine 13:
  github.com/minio/minio/cmd.(*ServerMux).handleServiceSignals()
      /home/vadmeste/work/gospace/src/github.com/minio/minio/cmd/service.go:86 +0x1fc

Goroutine 25 (running) created at:
  testing.(*T).Run()
      /home/vadmeste/work/go/src/testing/testing.go:646 +0x52f
  testing.RunTests.func1()
      /home/vadmeste/work/go/src/testing/testing.go:793 +0xb9
  testing.tRunner()
      /home/vadmeste/work/go/src/testing/testing.go:610 +0xc9
  testing.RunTests()
      /home/vadmeste/work/go/src/testing/testing.go:799 +0x4ba
  testing.(*M).Run()
      /home/vadmeste/work/go/src/testing/testing.go:743 +0x12f
  main.main()
      github.com/minio/minio/cmd/_test/_testmain.go:826 +0x1b8

Goroutine 13 (running) created at:
  github.com/minio/minio/cmd.(*ServerMux).ListenAndServe()
      /home/vadmeste/work/gospace/src/github.com/minio/minio/cmd/server-mux.go:307 +0x64
  github.com/minio/minio/cmd.TestListenAndServePlain.func2()
      /home/vadmeste/work/gospace/src/github.com/minio/minio/cmd/server-mux_test.go:234 +0x38
```

## Expected Behavior
No race detected

## Current Behavior
Race detected

## Steps to Reproduce (for bugs)
`go test -run=TestListenAndServePlain -race -count=100` inside github.com/minio/minio/cmd/

## Your Environment
Last master in ubuntu 16.04

**Comment by harshavardhana** [comment]
The bug is in the test itself, the problem is occurring in the way service channel is allocated with in the scope of the test itself.

**Comment by harshavardhana** [comment]
Closing this as this will be addressed in #3198
